### PR TITLE
API project - Add Owner_id and identifier feature

### PR DESCRIPTION
**Pull request by edsoubsoub**
Hi there,
We are using Kanboard to create projects/tasks automaticity from an external source, so we have an external id used to create and get them.
With this pull request your are able to create and update project with two more options :
 - owner_id
 - identifier

A new useful project API method getProjectByIdentifier has been had to app/Api/ProjectApi.php

Thx.

### CODE REVIEW

File: app/Api/ProjectApi.php [hunk 1]
```diff
@@ -62,26 +67,28 @@ public function getProjectActivity($project_id)
...
             'id' => $id,
             'name' => $name,
-            'description' => $description
+            'description' => $description,
+            'identifier' => $identifier
         );
 
         list($valid, ) = $this->projectValidator->validateModification($values);
-        return $valid && $this->projectModel->update($values);
+        return $valid && $this->projectModel->update($values, $owner_id);
```
Comment (fguillot) [inline app/Api/ProjectApi.php]: The method `ProjectModel::update()` do not have the argument `owner_id`. See https://github.com/fguillot/kanboard/blob/master/app/Model/ProjectModel.php#L429

File: app/Api/ProjectApi.php [hunk 2]
```diff
@@ -62,26 +67,28 @@ public function getProjectActivity($project_id)
...
+            'identifier' => $identifier
         );
 
         list($valid, ) = $this->projectValidator->validateCreation($values);
-        return $valid ? $this->projectModel->create($values) : false;
+        return $valid ? $this->projectModel->create($values, $owner_id) : false;
     }
 
-    public function updateProject($id, $name, $description = null)
+    public function updateProject($id, $name, $description = null, $owner_id=0,  $identifier=null)
```
Comment (fguillot) [inline app/Api/ProjectApi.php]: Be consistent with the existing code base, add spaces between the equals signs.

**Comment by edsoubsoub** [comment]
I've made the corrections.
Thx

**Comment by fguillot** [comment]
I merged manually your PR to fit with the latest changes in master.

For the next time, do not forget to update the documentation and update the integration tests.

Thanks

**Pull request closed without merging**